Assets/Scripts/UIStates/PauseMenuState.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class PauseMenuState : UIBaseState
+{
+    [SerializeField] private GameObject parentPauseMenu;
+    [SerializeField] private GameObject pauseScreen;
+    [SerializeField] private GameObject optionsScreen;
+    
+    [SerializeField] private UnityEvent onPause = new UnityEvent();
+    [SerializeField] private UnityEvent onUnpause = new UnityEvent();
+
+    private bool _isPaused;
+
+    protected override void EnterState()
+    {
+        UpdatePauseMenu();
+        TimeScaleController.SetTimeScale(0);
+    }
+
+    protected override void ExitState()
+    {
+        UpdatePauseMenu();
+        TimeScaleController.SetTimeScale(1);
+    }
+    
+    private void Start() 
+    {
+        if (pauseScreen) pauseScreen.SetActive(true);
+    }
+
+    private void UpdatePauseMenu()
+    {
+        TogglePauseMenu(_isPaused);
+        parentPauseMenu.SetActive(_isPaused);
+        optionsScreen.SetActive(false);
+        onUnpause?.Invoke();
+    }
+
+    public void TogglePauseMenu(bool isPaused)

```

Comment:
```suggestion
    public void SetPauseMenu(bool isPaused)
```
Je toggeld niet echt met deze functie, anders zal er geen parameter zijn. De naam kan dan beter "Set" gebruiken.

Anders zou het er zo uit zien
```suggestion
    public void TogglePauseMenu()
    {
        pauseScreen.SetActive(!_isPaused);
        _isPaused = !_isPaused;
    }
```